File: app/src/main/java/com/example/david/lists/view/addedit/userlist/AddEditUserListLogic.java
package com.example.david.lists.view.addedit.userlist;

import com.example.david.lists.data.datamodel.UserList;
import com.example.david.lists.data.repository.IRepositoryContract;
import com.example.david.lists.view.addedit.common.AddEditLogicBase;
import com.example.david.lists.view.addedit.common.IAddEditContract;

import static com.example.david.lists.view.addedit.common.IAddEditContract.TaskType.TASK_ADD;
import static com.example.david.lists.view.addedit.common.IAddEditContract.TaskType.TASK_EDIT;

public class AddEditUserListLogic extends AddEditLogicBase {

    public AddEditUserListLogic(IAddEditContract.View view,
                                IAddEditContract.ViewModel viewModel,
                                IRepositoryContract.Repository repository,
                                String id,
                                String title,
                                int position) {
        super(view, viewModel, repository, id, title, null, position);
    }


    @Override
    public void save(String newTitle) {
        if (viewModel.getCurrentTaskType() == TASK_ADD) {
            repository.addUserList(new UserList(newTitle, viewModel.getLastPosition()));
        } else if (viewModel.getCurrentTaskType() == TASK_EDIT) {
            repository.renameUserList(viewModel.getId(), newTitle);
        }
    }
}

File: app/src/main/java/com/example/david/lists/view/addedit/userlist/AddEditUserListLogic.kt
package com.example.david.lists.view.addedit.userlist

import com.example.david.lists.util.ISchedulerProviderContract
import com.example.david.lists.view.addedit.common.AddEditLogicBase
import com.example.david.lists.view.addedit.common.IAddEditContract
import com.example.david.lists.view.addedit.common.IAddEditContract.TaskType.ADD
import com.example.david.lists.view.addedit.common.IAddEditContract.TaskType.EDIT
import com.example.domain.repository.IRepositoryContract

class AddEditUserListLogic(view: IAddEditContract.View,
                           viewModel: IAddEditContract.ViewModel,
                           repo: IRepositoryContract.Repository,
                           schedulerProvider: ISchedulerProviderContract,
                           id: String,
                           title: String,
                           position: Int) :
        AddEditLogicBase(view, viewModel, repo, schedulerProvider, id, title, null, position) {

    public override fun save(newTitle: String) {
        when (viewModel.taskType) {
            ADD -> saveWithCompletable(repo.addUserList(newTitle, viewModel.position))
            EDIT -> saveWithCompletable(repo.renameUserList(viewModel.id, newTitle))
        }
    }
}
